src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -19,17 +23,34 @@ export default function CustomFieldInputSingleChoice(props) {
   />);
 
   function onClick() {
-    setShowOptions(true);
+    setShowOptions(!props.readOnly);
   }
 
   function onKeyUp(event) {
     if (event.key === 'Enter') {
-      setShowOptions(true);
+      setShowOptions(!props.readOnly);
     } else if (event.key === 'Escape') {
       setShowOptions(false);
     }
   }
 
+  const listOptions = props.choices.map((item, index) => (
+    <ListOption
+      dataId={item.id}
```

Comment:
We could potentially use the native `id` attribute, if we can guarantee that they'll be unique. This could be achieved by prepending something like `list-option-${props.id}` and making `ListOption` require an `id` param. How do you feel about that approach?

Reply:
Or, how about using the `value` attribute? Looks like we set `value` to the choice's label but a (single choice) field's value is actually the choice's `id`.